front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/formatSchedule.ts

```diff
@@ -32,9 +32,10 @@ const formatSchedule = (
         arrival: formatArrival ?? undefined,
         locked: step.locked,
         on_stop_signal: step.onStopSignal,
-        stop_for: isNaN(Number(step.stopFor))
-          ? undefined
-          : formatDurationAsISO8601(Number(step.stopFor)),
+        stop_for:
+          isNil(step.stopFor) || isNaN(Number(step.stopFor))
```

Comment:
Maybe you can just add this condition ?

```suggestion
          !step.stopFor
```

Reply:
I think we should keep `isNan`

Reply:
I think we should keep the isNan, it's to prevent bad behaviours from the user (if he tries to enter 'abc' in the cell for instance)